app/charts/index.ts

```diff
@@ -1444,15 +1450,32 @@ const chartConfigsAdjusters: ChartConfigsAdjusters = {
           return newChartConfig;
         },
       },
-      y: ({ newChartConfig, oldChartConfig, measures }) => {
+      y: ({ newChartConfig, oldChartConfig, measures, isAddingNewCube }) => {
         const numericalMeasures = measures.filter(isNumericalMeasure);
         const numericalMeasureIds = numericalMeasures.map((d) => d.id);
         let leftMeasure = numericalMeasures.find(
           (d) => d.id === numericalMeasureIds[0]
         ) as NumericalMeasure;
         let rightMeasureId: string | undefined;
-        const getMeasure = (id: string) => {
-          return numericalMeasures.find((d) => d.id === id) as NumericalMeasure;
+        const getLeftMeasure = (id: string) => {
+          if (isAddingNewCube) {
+            const rightMeasure = numericalMeasures.find(
+              (d) => d.id === rightMeasureId
+            );
+
+            // We're trying to find a measure that comes from a different cube,
+            // to better showcase the fact that we have two cubes now.
+            return (numericalMeasures.find(
+              (d) =>
+                d.cubeIri !== rightMeasure?.cubeIri &&
+                d.unit !== rightMeasure?.unit
+            ) ??
```

Comment:
Maybe restructuring this into variables before returning would help, and maybe even prevent the need for comments? - What do you think?

Reply:
Yes, good point – I tried to make it more clear in https://github.com/visualize-admin/visualization-tool/pull/1907/commits/f7ad94bc610a2de5db6857f494c7ca2e3ed35061 🥹